pkgs/top-level/all-packages.nix

```diff
@@ -35300,8 +35300,8 @@ with pkgs;
 
   angband = callPackage ../games/angband { };
 
-  anki = python39Packages.callPackage ../games/anki {
-    inherit (darwin.apple_sdk.frameworks) CoreAudio;
+  anki = callPackage ../games/anki {
+    # inherit (darwin.apple_sdk.frameworks) CoreAudio;
```

Comment:
Why not python39Packages.callPackage?

Also, why not regular python3?

Reply:
The use of `callPackage` is for two reasons:

1. The nixpkgs docs encourage it, i.e. here: https://github.com/NixOS/nixpkgs/blob/c39b1ec7c07feb5fd9fad0bacbe8f4b74b23ab06/doc/languages-frameworks/python.section.md?plain=1#L1158-L1160
2. Doing it naively results in `protobuf`, at the top, being the python protobuf package instead of the top-level `pkgs.protobuf`, so it seemed easier to follow the guide in that regard too

----

python39 is because upstream is bundling 3.9, not 3.10 or some other python: https://github.com/ankitects/anki/blob/76d8807315fcc2675e7fa44d9ddf3d4608efc487/build/ninja_gen/src/python.rs#L21-L26

My experience has been that python minor versions like 3.9 vs 3.10 often have enough differences that it's worth sticking to a known-to-work version.